### PR TITLE
Auto trait design note

### PR DESCRIPTION
**Pull request by Mark-Simulacrum**
I would like to iterate some more on this -- it feels a little scattered. But wanted to post it for initial feedback anyway, and maybe people have ideas on how to structure it better.

### CODE REVIEW

File: src/design_notes/auto_traits.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+Auto traits are tracked in [rust-lang/rust#13231], and are also sometimes
+referred to as OIBITs.
+
+As of November 2020, the language team feels that new auto traits are unlikely
+to be added or stabilized. See [discussion][freeze discussion] on the addition of `Freeze` for
+context. There is a fairly high burden to doing so on the ecosystem, as it
+becomes a concern of every library author whether to implement the trait or not.
+
+Each auto trait represents a semver compatability hazard for Rust libraries, as
```
Comment (joshtriplett) [inline src/design_notes/auto_traits.md:16]:
```suggestion
Each auto trait represents a semver compatibility hazard for Rust libraries, as
```

File: src/design_notes/auto_traits.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# Auto traits
+
+Auto traits permit automatically implementing a trait for types which contain
+fields implementing the trait. That is, they are fairly close to an automatic
+derive. They describe properties of types rather than behaviors; current stable
+Rust has two auto traits: `Send` and `Sync`.
+
+Auto traits are tracked in [rust-lang/rust#13231], and are also sometimes
+referred to as OIBITs.
```
Comment (joshtriplett) [inline src/design_notes/auto_traits.md:9]:
```suggestion
referred to as OIBITs ("opt-in built-in traits").
```

File: src/design_notes/auto_traits.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
+# Auto traits
+
+Auto traits permit automatically implementing a trait for types which contain
+fields implementing the trait. That is, they are fairly close to an automatic
+derive. They describe properties of types rather than behaviors; current stable
+Rust has two auto traits: `Send` and `Sync`.
```
Comment (cramertj) [inline src/design_notes/auto_traits.md:6]: + `Unpin` + `UnwindSafe`

Comment (Mark-Simulacrum) [reply src/design_notes/auto_traits.md:6]:
```
$ rg 'auto trait' library/
library/core/src/marker.rs
39:pub unsafe auto trait Send {
470:pub unsafe auto trait Sync {
717:pub(crate) unsafe auto trait Freeze {}
773:pub auto trait Unpin {}

library/rtstartup/rsbegin.rs
25:auto trait Sync {}
29:auto trait Freeze {}

library/rtstartup/rsend.rs
17:auto trait Freeze {}

library/std/src/panic.rs
107:/// panic). This trait is an auto trait, so it is automatically implemented for
141:pub auto trait UnwindSafe {}
159:pub auto trait RefUnwindSafe {}
```

Comment (nikomatsakis) [reply src/design_notes/auto_traits.md:6]: Freeze is an "impl detail" (but perhaps worth mentioning)

**Review by nikomatsakis — APPROVED** [review]

**Pull request merged**